### PR TITLE
FOUNDATIONS: Remove Score

### LINKED ISSUE #235: FOUNDATIONS: Remove `Score`

### PR DESCRIPTION
**Pull request by SardorSohinazarov**
closes #235

### CODE REVIEW

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.RemoveById.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+namespace Tarteeb.Api.Tests.Unit.Services.Foundations.Scores
+{
+    public partial class ScoreServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyValidationOnRemoveIfDatabaseUpdateConcurrencyErrorOccursAndLogItAsync()
+        {
+            //given
+            Guid someScoreId = Guid.NewGuid();
+
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.RemoveById.cs:25]: remove line

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.RemoveById.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+namespace Tarteeb.Api.Tests.Unit.Services.Foundations.Scores
+{
+    public partial class ScoreServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnRemoveIfIdIsInvalidAndLogItAsync()
+        {
+            //given
+            Guid invalidScoreId = Guid.Empty;
+
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.RemoveById.cs:23]: remove line

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.RemoveById.cs (hunk 1 above)
Comment (SardorSohinazarov) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.RemoveById.cs:25]: done

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.RemoveById.cs (hunk 2 above)
Comment (SardorSohinazarov) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.RemoveById.cs]: done

File: Tarteeb.Api/Services/Foundations/Scores/ScoreService.Exceptions.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+using Xeptions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Data.SqlClient;
+
+namespace Tarteeb.Api.Services.Foundations.Scores
+{
+    public partial class ScoreService
+    {
+        private delegate ValueTask<Score> ReturningScoresFunction();
+        private async ValueTask<Score> TryCatch(ReturningScoresFunction returningTeamFunction)
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Services/Foundations/Scores/ScoreService.Exceptions.cs:19]: Empty space

**Comment by ElbekDeveloper** [comment]
Very good. Just a few comments. Resolve them and you can merge it

**Review by ElbekDeveloper — APPROVED** [review]

**Pull request merged**